pybamm/expression_tree/binary_operators.py

```diff
@@ -827,7 +827,7 @@ def simplified_addition(left, right):
             )
         ) and all(
             left.evaluates_on_edges(dim) == right.evaluates_on_edges(dim)
-            for dim in ["primary", "secondary", "tertiary"]
+            for dim in ["primary", "secondary", "tertiary", "quaternary"]
```

Comment:
We should make `evaluates_on_edges("all")` check all the dimensions to avoid this repetitive construction

Reply:
This is checking that `evaluates_on_edges` match for left and right in each dimension (rather than all on edges or all on nodes), right? After playing around, keeping the domain loop outside of `evaluate_on_edges` seemed a lot cleaner, just replacing the list `["primary", "secondary", ...]` with `left.domains.keys()`. (Both `left` and `right` have the same domains at this point, I think? So can just use the domain list of one of them.)